### PR TITLE
Do not hard-fail on process owner check

### PR DESCRIPTION
**Pull request by zopieux**
`ps` might be missing in $PATH, or fan2go might be running as a non-root user with write access to the subsystem. I don't think it's reasonable to crash the program for a simple uid check that might actually be wrong. This will eventually crash later when trying to modify paths as non-root anyway.

### CODE REVIEW

**Comment by markusressel** [comment]
Hey, thx for opening the PR! :heart: 

There are definitely scenarios in which you do not need to use root permissions for running fan2go, so I can absolutely see why we need to improve the current logic :+1: 

> `ps` might be missing in $PATH, or fan2go might be running as a non-root user with write access to the subsystem. I don't think it's reasonable to crash the program for a simple uid check that might actually be wrong.

I agree, this code is a remnant from the early days and also my specific usage. However, I think we need to be a bit more specific here, since there is an additional permission related check done while evaluating the configuration file meant to protect users against malicious configuration editors. Did you consider this?

> This will eventually crash later when trying to modify paths as non-root anyway.

Did you validate this? Because I am not 100% sure what the current behavior would be :thinking:
When a fan cannot be controlled during normal operation, the error should _not_ lead to a crash, but it might do that when the error is already present during startup (of the fan2go service).

**Comment by zopieux** [comment]
>Did you validate this?

I did not, "crash" might be the wrong word here. I meant that I/O while trying to set the PWM will fail in a way or another and hopefully bubble up in the logs, so it'll be pretty obvious something's wrong.

>Did you consider this?

I'm not sure what malicious editor protection you're referring to here, sorry, I only skimmed at the surface of fan2go – thanks btw, what a pleasure it was jettisoning that fancontrol garbage!

I don't really mind *how* this check is done, I just believe depending on a subprocess invocation & a (very common but not necessarily available) Linux util is overkill for a simple user warning message. Maybe Go has a way to report its own uid, [like this](https://pkg.go.dev/os/user#Current)?

Feel free to close this PR and come up with something better :)

**Comment by markusressel** [comment]
> I don't really mind _how_ this check is done, I just believe depending on a subprocess invocation & a (very common but not necessarily available) Linux util is overkill for a simple user warning message. Maybe Go has a way to report its own uid, [like this](https://pkg.go.dev/os/user#Current)?

Good find! Seems like a much more reasonable approach :+1: 

> Feel free to close this PR and come up with something better :)

I think we can use it for now, I will think about other special cases in a later PR.

File: internal/backend.go [hunk 1]
```diff
@@ -18,18 +18,20 @@ import (
...
 
 func RunDaemon() {
-	if owner, err := getProcessOwner(); err != nil && owner != "root" {
-		ui.Fatal("Fan control requires root permissions to be able to modify fan speeds, please run fan2go as root")
+	owner, err := getProcessOwner()
+	if err != nil {
+		ui.Warning("Unable to verify process owner: %v", err)
+	}
+	if owner != "root" {
+		ui.Info("fan2go is running as a non-root user '%s'. If you encounter errors, make sure to give this user the required permissions.", owner)
```
Comment (zopieux) [inline internal/backend.go:33]: Nice! Making it non-fatal makes a lot of sense actually.

**Pull request merged**